lib/constellation.rb

```diff
@@ -1,58 +1,31 @@
-require 'net/http'
+require "graphql/client"
+require "graphql/client/http"
 module Constellation
-  class HttpException < StandardError
+  class GraphQLException < StandardError
   end
-  HTTP_ERRORS = [
-    Errno::ETIMEDOUT,
-    Timeout::Error,
-    Net::OpenTimeout,
-    Net::ReadTimeout,
-    EOFError,
-    Errno::ECONNABORTED,
-    Errno::ECONNREFUSED,
-    Errno::ECONNRESET,
-    Errno::EHOSTDOWN,
-    Errno::EHOSTUNREACH,
-    Errno::EINVAL,
-    Errno::ENETUNREACH,
-    SocketError,
-    OpenSSL::SSL::SSLError,
-    Net::HTTPBadResponse,
-    Net::HTTPHeaderSyntaxError,
-    Net::ProtocolError,
-    Zlib::GzipFile::Error,
-  ]
 
-  def self.create_rsvp!(rsvp_params)
-    return unless constellation_enabled?
-    uri = URI.parse("#{app}/rsvps")
-    http = Net::HTTP.new(uri.host, uri.port)
-    http.use_ssl = true
-    req = Net::HTTP::Post.new(uri.path, headers)
-    req.body = { rsvp: rsvp_params.merge(total_count: true) }.to_json
-    response = http.request(req)
-    raise Constellation::HttpException, response.body unless response.code == '201'
-    return JSON.parse(response.body), response['X-Total-Count'].to_i
-  rescue *HTTP_ERRORS => e
-    raise Constellation::HttpException, e.message
-  end
-
-  def self.constellation_enabled?
-    app && token
+  HTTP = GraphQL::Client::HTTP.new("#{RsvpRails.config[:constellation_url]}/api/graphql/") do
+    def headers(_)
+      { 'Authorization' => "Bearer #{RsvpRails.config[:constellation_jwt_token]}" }
+    end
   end
+  Schema = GraphQL::Client.load_schema(HTTP)
+  Client = GraphQL::Client.new(schema: Schema, execute: HTTP)
 
-  def self.app
-    RsvpRails.config[:constellation_url]
-  end
+  CreateRsvpMutation = Client.parse <<-GRAPHQL
+    mutation($input: createRsvpInput!) {
+      createRsvp(input: $input) {
+        id
+        total_count
+      }
+    }
+  GRAPHQL
 
-  def self.token
-    RsvpRails.config[:jwt_token]
-  end
+  def self.create_rsvp!(rsvp_params)
+    rsvp_params[:event_id] = rsvp_params[:event_id].to_s
```

Comment:
sort of silly, maybe we should change this in Constellation to require Int not String?

Reply:
I think these are `ID!`s in Constellation side, and I believe those default to String. +1 on changing them to `Int!`, at least some projects out there do it.

Reply:
I think it should be a `GraphQLID`, which actually still serialises as a string http://facebook.github.io/graphql/October2016/#sec-ID